x/evm/keeper/abci.go

```diff
@@ -31,6 +31,13 @@ func (k *Keeper) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 	// reset counters that are used on CommitStateDB.Prepare
 	k.Bloom = big.NewInt(0)
 	k.TxCount = 0
+
+	//loading revision id from store in every block, make sure the `csdb.nextRevisionId` consistent in every node
+	id, err := k.GetRevisionID(ctx)
+	if err != nil {
+		panic(err)
+	}
+	k.CommitStateDB.SetRevisionID(id)
```

Comment:
I don't understand why you need to get the get the id from the store to then set it to the CSDB. Why not set it only during EndBlock using the KVStore that is already available on the CSDB?

Reply:
Because of we must make sure the `revision id` consistent in any state of node, the state including:
* running node
* restarted node with snapshot(start with snapshot's height, sync missing blocks from other node)
* restarted node without snapshot(start with height 0, sync all blocks from other node)
for situation 2, we don't executing all transactions , so we can not sync `revision id` with others by executing TX, so we need read it in `BeginBlock` to make sure the `revision id` consistent with other running node during executing the synchronized TX in block

Reply:
> For your point, did you means that we should setting revision id to store when nextRevisionID changed, and getting revision id from store when we need a nextRevisionID ?

yeah that's what I meant.

Reply:
i have finished change by your point